base_mixin_restrict_field_access/models/restrict_field_access_mixin.py

```diff
@@ -0,0 +1,312 @@
+                  self._restrict_field_access_suspend())
+            .create(restricted_vals_list).ids
+        )
+
+    @api.multi
+    def copy(self, default=None):
+        restricted_default = self._restrict_field_access_filter_vals(
+            default or {}, action='create')
+        return self.browse(
+            super(RestrictFieldAccessMixin,
+                  self._restrict_field_access_suspend())
+            .copy(default=restricted_default).ids
+        )
+
+    @api.multi
+    def read(self, fields=None, load='_classic_read'):
+        result = super(RestrictFieldAccessMixin, self).read(
+            fields=fields, load=load)
+        for record in result:
+            this = self.browse(record['id'])
+            for field in record:
+                if not this._restrict_field_access_is_field_accessible(field):
+                    record[field] = self._fields[field].convert_to_read(
+                        self._fields[field].null(self), record)
+                    if self.env.cache.get(self, self._fields[field]):
+                        self.env.cache.remove(self, self._fields[field])
+        return result
+
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None,
+                   orderby=False, lazy=True):
+        """Restrict reading if we read an inaccessible field"""
+        has_inaccessible_field = False
+        has_inaccessible_field |= any(
+            not self._restrict_field_access_is_field_accessible(f)
+            for f in fields or self._fields.keys()
+        )
+        has_inaccessible_field |= any(
+            expression.is_leaf(term) and
+            not self._restrict_field_access_is_field_accessible(
+                term[0].split('.')[0]
+            )
+            for term in domain
+        )
+        if groupby:
+            if isinstance(groupby, str):
+                groupby = [groupby]
+            has_inaccessible_field |= any(
+                not self._restrict_field_access_is_field_accessible(
+                    f.split(':')[0]
+                )
+                for f in groupby
+            )
+        if orderby:
+            has_inaccessible_field |= any(
+                not self._restrict_field_access_is_field_accessible(f.split())
+                for f in orderby.split(',')
+            )
+        # just like with search, we restrict read_group to the accessible
+        # records, because we'd either leak data otherwise or have very wrong
+        # results
+        if has_inaccessible_field:
+            self._restrict_field_access_inject_restrict_field_access_domain(
+                domain
+            )
+
+        return super(RestrictFieldAccessMixin, self).read_group(
+            domain, fields, groupby, offset=offset, limit=limit,
+            orderby=orderby, lazy=lazy
+        )
+
+    @api.multi
+    def _export_rows(self, fields, *, _is_toplevel_call=True):
```

Comment:
It was `_BaseModel__export_rows`, I don't know what's the difference between this call and the the direct call.
Note (don't know if it's related): in v8 it was `__export_rows` (2 _), and not `_export_rows`

cc @hbrunn

Reply:
that was some inheritance-foo in v8, if this works just like that in v12 then we're happy